timely_beliefs/visualization/utils.py

```diff
@@ -294,3 +300,103 @@ def align_belief_horizons(
     df2 = df.copy().iloc[0:0]
     df2 = df2.append(pd.DataFrame(data, columns=df2.columns))
     return df2
+
+
+def ridgeline_plot(
+    bdf,
+    fixed_viewpoint: bool = False,
+    distribution: str = "uniform",
+    event_value_window: Tuple[float, float] = None,
+) -> alt.FacetChart:
+    """
+    Creates ridgeline plot
+
+    :param bdf: BeliefsDataFrame
+    :param fixed_viewpoint: boolean, if true create fixed viewpoint plot
+    :param distribution: string, distribution name to use (discrete, normal or uniform)
+    :param event_value_window: optional tuple specifying an event value window for the x-axis
+           (e.g. plot temperatures between -1 and 21 degrees Celsius)
+    """
+    df = interpret_and_sample_distribution_long_form(
+        bdf, distribution=distribution, event_value_window=event_value_window
+    ).reset_index()
+    df["belief_horizon"], belief_horizon_unit = timedelta_to_human_range(
+        df["belief_horizon"]
+    )
+
+    step = 10
+    overlap = 50
+    probability_scale_range = (step, -step * overlap)
+
+    deterministic_chart = graphs.deterministic_chart(probability_scale_range)
+    probabilistic_chart = graphs.probabilistic_chart(
+        probability_scale_range,
+        belief_horizon_unit=belief_horizon_unit,
+        sensor_name=bdf.sensor.name,
+        sensor_unit=bdf.sensor.unit,
+    )
+    ridgeline_chart = (
+        alt.layer(
+            probabilistic_chart,
+            deterministic_chart,
+            selectors.ridgeline_selector(probability_scale_range, belief_horizon_unit),
+            data=df,
+        )
+        .properties(height=step)
+        .facet(
+            row=alt.Row(
+                "belief_horizon:N",
+                sort="descending",
+                title=("Upcoming " if fixed_viewpoint else "Previous ")
+                + belief_horizon_unit,
+                header=alt.Header(
+                    labelAngle=0, labelAlign="right", format="%B"
+                ),  # todo: set conditional labels once labelExpr finds its way from vega-lite to altair,
+                #      so then we can choose to print e.g. 0, 6, 12, 18, 24 hours instead of all of 0 to 24 hours.
+                #      See https://github.com/vega/vega-lite/issues/5310
+            )
+        )
+        .properties(bounds="flush")
+        .configure_facet(spacing=0)
+        .configure_view(stroke=None)
+        .configure_title(anchor="end")
+    )
+
+    return ridgeline_chart
+
+
+def interpret_and_sample_distribution_long_form(
+    df: "classes.BeliefsDataFrame",
+    distribution: str = "uniform",
+    event_value_window: Tuple[float, float] = None,
+) -> pd.DataFrame:
+    """Interpret probabilistic beliefs as a continuous or discrete distribution of possible outcomes.
+    Return a DataFrame with a column for each belief containing probabilities of a given outcome (i.e. the index).
```

Comment:
You're using a function called `drawPDF` to generate this data. Can you explain why and where that function lives?